tests/helpers/execute.ts

```diff
@@ -51,9 +51,9 @@ export const executeCli = async (args: string[] = [], opts: { env: Env } = { env
   const promise = new Promise<ExecuteReturn>((resolve, reject) => {
     const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
     childProcess.stdin.setDefaultEncoding('utf-8');
-    let stdout!: string, stderr!: string;
-    const stdoutPromise = readStream(childProcess.stdout).then((value: undefined | string) => (stdout = value ?? ''));
-    const stderrPromise = readStream(childProcess.stderr).then((value: undefined | string) => (stderr = value ?? ''));
+    let stdout: string, stderr: string;
```

Comment:
i think the code is clearer when not using variables here. the ui doesn't let me put a suggestion so i just pasted the code.
```typescript
async function handleProcessExitEvent(stderrPromise: Promise<string>, stdoutPromise: Promise<string>): Promise<{ stdout: string; stderr: string }> {
  const stderr = await stderrPromise;
  const stdout = await stdoutPromise;
  return { stderr, stdout };
}

export const executeCli = async (args: string[] = [], opts: { env: Env } = { env: undefined }): Promise<ExecuteReturn> => {
  const { env = undefined } = opts;
  const promise = new Promise<ExecuteReturn>((resolve, reject) => {
    const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
    childProcess.stdin.setDefaultEncoding('utf-8');
    
    const stdoutPromise = readStream(childProcess.stdout);
    const stderrPromise = readStream(childProcess.stderr);

    childProcess.once('exit', (code) => {
      handleProcessExitEvent(stderrPromise, stdoutPromise)
        .then(({ stderr, stdout }) => {
          resolve({ exitCode: code, stdout, stderr });
        })
        .catch(() => {
          reject('read streams failed');
        });
    });
    childProcess.on('error', reject);
  });

  return promise;
};
```

Reply:
fixed